### PR TITLE
nodejs-week1

### PR DESCRIPTION
**Pull request by Sana-Shabeel**

### CODE REVIEW

**Comment by shpomp** [comment]
Will review in the upcoming days :)

File: nodejs/week1/homework/warmup.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    return this.radius * 2;
+  }
+  getCircumference() {
+    return this.radius * 2 * Math.PI;
+  }
+  getArea() {
+    return Math.PI * this.radius ** 2;
+  }
+}
+
```
Comment (shpomp) [inline nodejs/week1/homework/warmup.js:19]: Clean, clear, correct. 👌🏽  👏🏽  ✨ 

Consideration for improvements:

- We are getting some crazy fractions for circumference and area. Perhaps would make sense to round the numbers? 
- `Math.PI * this.radius ** 2` is perfectly fine, but for readability and practice `Math.pow()` could be tried out to perform one of the operations here.

**Comment by shpomp** [comment]
Good job overall!  👏🏽 
However, the reviews and the random fetches threw this off a little. 
Try to fix it and message me if any questions!
For the fetching of the random meals/reservations, the tets are misleading as they pass, while they should fail. I will inform somebody who can correct that :)

**Review by shpomp — CHANGES_REQUESTED** [review]

File: nodejs/week1/homework/src/backend/app.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+app.get("/meal", async (request, response) => response.send(meals.id));
+
+app.get("/reservations", async (request, response) =>
+  response.send(reservations)
+);
+app.get("/reservation", async (request, response) =>
+  response.send(reservations.id)
+);
+
```
[the remark this answers is not in the document]
Comment (Sana-Shabeel) [reply nodejs/week1/homework/src/backend/app.js:31]: Roger that. I will make the changes tomorrow

[the remark this answers is not in the document]
Comment (Sana-Shabeel) [reply nodejs/week1/homework/src/backend/app.js:49]: i made the changes you recommended.

File: nodejs/week1/homework/src/backend/app.js [hunk 3]
```diff
@@ -4,29 +4,54 @@ const app = express();
...
 
-// this is where you will be adding your routes
-app.get("/", async (request, response) =>
-  response.send("Meal Sharing Web App")
-);
+//  meals with reviews
+meals.forEach((meal, idx) => {
+  const review = reviews.filter((res) => res.mealId === meal.id);
+  meal.review = review;
+});
```
Comment (shpomp) [inline nodejs/week1/homework/src/backend/app.js:13]: meal.reviews

Comment (Sana-Shabeel) [reply nodejs/week1/homework/src/backend/app.js:13]: DONE. 
can i merge now ?

**Pull request merged**